system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -5354,6 +5365,8 @@ class CLdapClient : implements ILdapClient, public CInterface
         while(nextptr != NULL)
         {
             prevptr = nextptr + 2;
+            if (prevptr == nullptr)
+                break;//could happen if resource contains trailing ::
```

Comment:
How? *prevptr could be '\0', but prevptr cannot be null (unless nextptr was "null-2")